packages/twenty-front/src/modules/object-record/record-field/form-types/components/FormTextFieldInput.tsx

```diff
@@ -0,0 +1,86 @@
+import { StyledFormFieldInputContainer } from '@/object-record/record-field/form-types/components/StyledFormFieldInputContainer';
+import { StyledFormFieldInputInputContainer } from '@/object-record/record-field/form-types/components/StyledFormFieldInputInputContainer';
+import { StyledFormFieldInputRowContainer } from '@/object-record/record-field/form-types/components/StyledFormFieldInputRowContainer';
+import { TextVariableEditor } from '@/object-record/record-field/form-types/components/TextVariableEditor';
+import { useTextVariableEditor } from '@/object-record/record-field/form-types/hooks/useTextVariableEditor';
+import { VariablePickerComponent } from '@/object-record/record-field/form-types/types/VariablePickerComponent';
+import { InputLabel } from '@/ui/input/components/InputLabel';
+import { parseEditorContent } from '@/workflow/search-variables/utils/parseEditorContent';
+import { useId } from 'react';
+import { isDefined } from 'twenty-ui';
+
+type FormTextFieldInputProps = {
+  label?: string;
+  defaultValue: string | undefined;
+  placeholder: string;
+  onPersist: (value: string) => void;
+  multiline?: boolean;
+  readonly?: boolean;
+  VariablePicker?: VariablePickerComponent;
+};
+
+export const FormTextFieldInput = ({
+  label,
+  defaultValue,
+  placeholder,
+  onPersist,
+  multiline,
+  readonly,
+  VariablePicker,
+}: FormTextFieldInputProps) => {
+  const inputId = useId();
+
+  const editor = useTextVariableEditor({
+    placeholder,
+    multiline,
+    readonly,
+    defaultValue,
+    onUpdate: (editor) => {
+      const jsonContent = editor.getJSON();
+      const parsedContent = parseEditorContent(jsonContent);
+
+      onPersist(parsedContent);
+    },
+  });
+
+  const handleVariableTagInsert = (variable: string) => {
```

Comment:
Could we avoid using just "variable" alone, either `variableValue`, `variableContent` or `variableName` ? As a first-time reader I don't clearly understand, which for this "meta" part of the code is important.

Reply:
Yes 👌 Done.